api/v1alpha2/ipamprovider_types.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2023 The Kubernetes Authors.
```

Comment:
nit: header year has to be updated here in other places

Reply:
Idk... the code was written last year and hasn't been updated after, so I think it is correct to put 2023 there.